### PR TITLE
feat: FAB container and add back to top fab

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 8218066</samp>

This pull request adds a feature to the site layout that allows users to scroll back to the top of the page with a floating action button (FAB). It introduces new components `BackToTopFAB`, `FABBase`, and `FABContainer`, as well as a new module `FAB.tsx` that contains the FAB logic and context.

### <samp>🤖 Generated by Copilot at 8218066</samp>

> _To improve the user's navigation_
> _We added a new FAB creation_
> _Using `FABBase` and `FABContainer`_
> _And a custom SVG `BxBxsArrowToTop` render_
> _We made a smooth scroll to top animation_

### <samp>🤖 Generated by Copilot at 8218066</samp>

*  Add a new component `BackToTopFAB` that renders a FAB with an arrow icon to scroll to the top ([link](https://github.com/Crossbell-Box/xLog/pull/415/files?diff=unified&w=0#diff-eba52c8b223a7e6bed46fd053fa63e7bfcee59a2ba8a502510a598ac00700043R1-R24))
*  Import and use `FABContainer` and `BackToTopFAB` in `SiteLayout` to display the FAB conditionally based on the scroll position ([link](https://github.com/Crossbell-Box/xLog/pull/415/files?diff=unified&w=0#diff-43426959eff98a692b60240b811291a96f1325bd39fc047cf10e89b32ce18513R17-R18), [link](https://github.com/Crossbell-Box/xLog/pull/415/files?diff=unified&w=0#diff-43426959eff98a692b60240b811291a96f1325bd39fc047cf10e89b32ce18513R112-R119))
*  Create and export `FABBase` and `FABContainer` components, as well as `FABContainerContext` interface and `useFabContextState` hook in `FAB.tsx` ([link](https://github.com/Crossbell-Box/xLog/pull/415/files?diff=unified&w=0#diff-5a88abb6f102cf84869d4f48d055631ec1dd991b08053bc7f774707845311e5fR1-R55))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Comment by Innei** [comment]
Feat: 

- add hook `useFAB` to dynamic add or remove FAB.
- add back to top fab.




Demo:

https://user-images.githubusercontent.com/41265413/233841480-7a898e0e-472c-4a02-a280-1f6d99827791.mov

File: src/components/site/BackToTopFAB.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+const isShouldShow = () =>
+  document.documentElement.scrollTop > document.documentElement.clientHeight
+
+export const BackToTopFAB: React.FC<{}> = () => {
+  const [shouldShow, setShouldShow] = useState(isShouldShow())
+  useEffect(() => {
+    const handler = () => {
+      setShouldShow(isShouldShow())
```
Comment (dohooo) [inline src/components/site/BackToTopFAB.tsx:20]: Hmm, maybe we don't need to listen to the scroll changes immediately. So, we can add a throttle wrapper to this function to reduce the state changes.

File: src/components/ui/FAB.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    >
+      {children}
+    </button>
+  )
+}
+
+export const FABContainer: FC<{
+  children: JSX.Element | JSX.Element[]
+}> = (props) => {
+  const [fabConfig, setFabConfig] = useState([] as FABConfig[])
```
Comment (dohooo) [inline src/components/ui/FAB.tsx:109]: IMHO.
```ts
const [fabConfig, setFabConfig] = useState<FABConfig[]>([]);
```

**Review by dohooo — CHANGES_REQUESTED** [review]
This is an excellent PR, and now we can easily add FAB in the future! Thank you for your effort! 🫡

**Review by dohooo — APPROVED** [review]

**Comment by DIYgod** [comment]
The mingcute website has the icon `arow_to_up_line`, but its repository hasn't been updated. I will keep an eye on it. https://github.com/Richard9394/MingCute/issues/13

**Pull request merged**